src/interned.rs

```diff
@@ -137,7 +137,25 @@ where
 
         // Optimisation to only get read lock on the map if the data has already
         // been interned.
```

Comment:
We should add a comment here about how we have to use the raw API because we want to avoid creating an owned copy of the data if not necessary and how the `Borrow` trait isn't flexible enough for this use case (it requires that an `&self` be converted to `&Borrow` but we have a struct that wraps a number of fields).